homeassistant/components/comfoconnect/fan.py

```diff
@@ -89,41 +94,36 @@ def supported_features(self) -> int:
         return SUPPORT_SET_SPEED
 
     @property
-    def speed(self):
-        """Return the current fan mode."""
-        try:
-            speed = self._ccb.data[SENSOR_FAN_SPEED_MODE]
-            return SPEED_MAPPING[speed]
-        except KeyError:
+    def percentage(self) -> str:
+        """Return the current speed percentage."""
+        speed = self._ccb.data[SENSOR_FAN_SPEED_MODE]
+        if speed is None:
             return None
+        return ranged_value_to_percentage(SPEED_RANGE, speed)
 
-    @property
-    def speed_list(self):
-        """List of available fan modes."""
-        return [SPEED_OFF, SPEED_LOW, SPEED_MEDIUM, SPEED_HIGH]
-
-    def turn_on(self, speed: str = None, **kwargs) -> None:
+    def turn_on(
+        self, speed: str = None, percentage=None, preset_mode=None, **kwargs
+    ) -> None:
         """Turn on the fan."""
-        if speed is None:
-            speed = SPEED_LOW
-        self.set_speed(speed)
+        self.set_percentage(percentage)
 
     def turn_off(self, **kwargs) -> None:
         """Turn off the fan (to away)."""
-        self.set_speed(SPEED_OFF)
-
-    def set_speed(self, speed: str):
-        """Set fan speed."""
-        _LOGGER.debug("Changing fan speed to %s", speed)
-
-        if speed == SPEED_OFF:
-            self._ccb.comfoconnect.cmd_rmi_request(CMD_FAN_MODE_AWAY)
-        elif speed == SPEED_LOW:
-            self._ccb.comfoconnect.cmd_rmi_request(CMD_FAN_MODE_LOW)
-        elif speed == SPEED_MEDIUM:
-            self._ccb.comfoconnect.cmd_rmi_request(CMD_FAN_MODE_MEDIUM)
-        elif speed == SPEED_HIGH:
-            self._ccb.comfoconnect.cmd_rmi_request(CMD_FAN_MODE_HIGH)
+        self.set_percentage(0)
+
+    def set_percentage(self, percentage: int):
+        """Set fan speed percentage."""
+        _LOGGER.debug("Changing fan speed percentage to %s", percentage)
+
+        if percentage is None:
+            cmd = CMD_FAN_MODE_LOW
```

Comment:
in this case would it not be more fair to put the comfoair in auto fan mode? I understand that this is complicated.

Alternative, I would set it to medium. (That is what my Auto defaults to in 95% of all cases)

Reply:
This is the existing behavior

https://github.com/home-assistant/core/pull/45593/files#diff-68160764433f2567b3118c66f3a5c55dfefd01cfabb001d7e7311c30472d2e83L116-L117